frontend_lib/src/component/Timeline/Timeline.jsx

```diff
@@ -9,40 +9,50 @@ import i18n from '../../i18n.js'
 import DisplayState from '../DisplayState/DisplayState.jsx'
 import { ROLE } from '../../helper.js'
 import { CUSTOM_EVENT } from '../../customEvent.js'
+import { TracimComponent } from '../../tracimComponent.js'
 
 // require('./Timeline.styl') // see https://github.com/tracim/tracim/issues/1156
 const color = require('color')
 
-class Timeline extends React.Component {
+export class Timeline extends React.Component {
   constructor (props) {
     super(props)
-    document.addEventListener(CUSTOM_EVENT.APP_CUSTOM_EVENT_LISTENER, this.customEventReducer)
+    props.registerCustomEventHandlerList([
+      { name: CUSTOM_EVENT.ALL_APP_CHANGE_LANGUAGE, handler: this.handleAllAppChangeLanguage }
+    ])
+
+    this.timelineContainerScrollHeight = 0
```

Comment:
Is there a reason this variable can't be a state?

Reply:
Yes, it isn't in the state because it is the result of a sum from 2 `ref` property value and it doesn't affect the render of the component so if this value has changed the component doesn't need to be re-rendered.